tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/AppFilterSuite.scala

```diff
@@ -528,4 +528,146 @@ class AppFilterSuite extends FunSuite {
       }
     }
   }
+
+  case class TestConjunctionAndDisjunction(
+      fileName: String, fsTime: Long, appName: String,
+      appTime: Long, uniqueId: Int, userName: String)
+
+  private val appsNameConjunctionAndDisjunctionToTest = Array(
+    TestConjunctionAndDisjunction("app-ndshours18", msHoursAgo(16), "Ndshours18",
+      msHoursAgo(18), 1, "user1"),
+    TestConjunctionAndDisjunction("app-ndsweeks-1", msWeeksAgo(1), "ndsweeks",
+      msWeeksAgo(1), 1, "user1"),
+    TestConjunctionAndDisjunction("app-ndsweeks-2", msWeeksAgo(2), "Ndsweeks",
+      msWeeksAgo(2), 2, "user2"),
+    TestConjunctionAndDisjunction("app-ndsweeks-3", msWeeksAgo(3), "ndsweeks",
+      msWeeksAgo(3), 3, "user3"),
+    TestConjunctionAndDisjunction("app-nds86-1", msDaysAgo(3), "nds86",
+      msDaysAgo(4), 1, "user1"),
+    TestConjunctionAndDisjunction("app-nds86-2", msDaysAgo(6), "nds86",
+      msWeeksAgo(1), 2, "user2"),
+    TestConjunctionAndDisjunction("app-nds86-3", msDaysAgo(18), "nds86",
+      msWeeksAgo(3), 3, "user3"))
+
+  test("Test disjunction all filters") {
+    testConjunctionAndDisjunction(appsNameConjunctionAndDisjunctionToTest,
+      filterCriteria("10-newest") ++ filterAppName("nds") ++
```

Comment:
nds app name matches everthing make it more unique

Reply:
nds app name matches with 5 out of 7 input.  Rest 2 are `Nds`. It is not doing Regex match in this case.  Could you please clarify what you mean by more unique.

Reply:
thats fine, I missed the Nds